packages/algob/test/internal/deployer.ts

```diff
@@ -419,4 +419,22 @@ describe("DeployerDeployMode", () => {
       deleted: false
     }]]));
   });
+
+  it("Should deploy asa without using asa.yaml", async () => {
+    const deployer = new DeployerDeployMode(deployerCfg);
+    const asaDef = {
+      total: 10000,
+      decimals: 0,
+      defaultFrozen: false,
+      unitName: "SLV",
+      url: "url",
+      metadataHash: "12312442142141241244444411111133",
+      note: "note"
+    };
+    await deployer.deployASADef("silver-122", asaDef, { creator: deployer.accounts[0] });
+
+    const res = deployer.getASAInfo("silver-122");
+    assert.isDefined(res);
+    assert.deepEqual(res.assetDef, asaDef);
+  });
```

Comment:
Are we also checking for required parameters, when user is passing `asaDef` in `deployASADef` (eg. `unitName`, `total` etc). Could you please add a test for that?

Reply:
yes, they are already checked because here we are not loading from file, we have types , so if we pass Ex: decimals as string it will fail at compilation.

Reply:
added parseDef and validate account